xcube/core/store/store.py

```diff
@@ -123,93 +123,118 @@ def get_data_store_params_schema(cls) -> JsonObjectSchema:
+        Get the tuple of data type specifiers that are supported for the given *data_id*.
+        In case the type specifier allows one ore more flags, they are listed in brackets
+        following the specifier's name, e.g., dataset[CUBE, MULTILEVEL].
+        :param data_id: An identifier of data that is provided by this store
+        :return: A tuple of type specifiers that apply to the given data_id
+        :raise DataStoreError: If an error occurs.
         """
-        Get an iterator over the data resource identifiers for the given type *type_id*.
-        If *type_id* is omitted, all data resource identifiers are returned.
 
-        If a store implementation supports only a single data type, it should verify that *type_id* is either None
-        or equal to that single data type.
+    @abstractmethod
+    def get_data_ids(self, type_specifier: str = None, include_titles = True) -> Iterator[Tuple[str, Optional[str]]]:
+        """
+        Get an iterator over the data resource identifiers for the given type *type_specifier*.
+        If *type_specifier* is omitted, all data resource identifiers are returned.
+
+        If a store implementation supports only a single data type, it should verify that *type_specifier*
+        is either None or equal to that single data type.
 
         The returned iterator items are 2-tuples of the form (*data_id*, *title*), where *data_id*
         is the actual data identifier and *title* is an optional, human-readable title for the data.
+        If *include_titles* is false, the second item of the result tuple will be None.
 
+        :param type_specifier: If given, only data identifiers that are available as this type are returned. If this is
+        omitted, all available data identifiers are returned
+        :param include_titles: If true, the store will attempt to also provide a title
         :return: An iterator over the identifiers and titles of data resources provided by this data store.
         :raise DataStoreError: If an error occurs.
         """
 
     @abstractmethod
-    def has_data(self, data_id: str) -> bool:
+    def has_data(self, data_id: str, type_specifier: str = None) -> bool:
         """
         Check if the data resource given by *data_id* is available in this store.
+
+        :param data_id: A data identifier
+        :param type_specifier: An optional data type specifier. If given, it will also be checked
+        whether the data is available as the specified type
         :return: True, if the data resource is available in this store, False otherwise.
         """
 
     @abstractmethod
-    def describe_data(self, data_id: str) -> DataDescriptor:
+    def describe_data(self, data_id: str, type_specifier: str = None) -> DataDescriptor:
         """
         Get the descriptor for the data resource given by *data_id*.
 
-        Raises if *data_id* does not exist in this store.
+        Raises a DataStoreError if *data_id* does not exist in this store
```

Comment:
```suggestion
        Raises a :class:DataStoreError if *data_id* does not exist in this store
```

Not clear yet when we raise class:DataStoreError  and when ValueError. For example in some modules, we raise ValueError if `type_specifier`is not compatible. That doesn't seem consistent to me.